lib/model/folder_sendrecv.go

```diff
@@ -798,15 +840,19 @@ func (f *sendReceiveFolder) deleteFile(file protocol.FileInfo, scanChan chan<- s
 		return dbUpdateJob{}, err
 	}
 
+	if cur.IsDirectory() {
+		return dbUpdateJob{}, errUnexpectedDirOnFileDel
+	}
```

Comment:
Why doesn't this trigger a scan?

Reply:
Is the added comment clear?

Reply:
Sure, but why are we not scheduling a scan in this case?

Reply:
So my comment wasn't clear :)  
Just above we checked that what is on disk (a dir) is the same as what is in the db, so a scan would not change anything about the situation. What is wrong is that the remote says: "Delete this file" and we see "We don't have a file, it's a dir". Maybe we could do some magic at this point to resolve the situation, but really I don't think this case should ever happen. However I can't prove it, so I want to make it a visible error, but not a panic, as that's really user-unfriendly. If I get proven wrong by reports, we can think about what magic is needed to resolve the situation.